magda-web-client/src/Components/DistributionRow.tsx

```diff
@@ -166,7 +167,24 @@ class DistributionRow extends Component<PropType> {
                         >
                             <img src={downloadIcon} alt="download" /> Download
                         </a>
-                    )}
+                    )}{" "}
+                    <br />
+                    <br />
+                    {distribution.ckanResource &&
+                        distribution.ckanResource.datastore_active && (
+                            <a
+                                className="download-button au-btn au-btn--secondary"
+                                target="_blank"
+                                rel="noopener noreferrer"
+                                href={distribution.sourceDetails.url.replace(
```

Comment:
This makes me a bit nervous, is there any chance at all that `distribution.sourceDetails` or `distribution.sourceDetails.url` could be null or undefined? I'd be a bit more comfortable if there was a check to avoid a nullpointererror whiting out the whole page 😬

Reply:
lodash _.get() to the rescue! I can use the generic CKAN API docs for the link if this field isn't there.